flag/flag.go

```diff
@@ -139,118 +152,172 @@ func (s *Set) parse() error {
 	return nil
 }
 
-func (s *Set) registerFieldFlag(name string, idxs []int, fieldVal reflect.Value, sf reflect.StructField) error {
-	ft := sf.Type
-	help := DefaultFlagHelpText
-	if x, ok := sf.Tag.Lookup(HelpTextTag); ok {
-		help = x
+func (s *Set) registerFlags(ptyp reflect.Type) error {
```

Comment:
Hmm, I missed that we've lost the value of the original template in this function call before.

That template value was used for setting the default values on the flags below (mostly for documentation when someone ones a command with `--help`).
It looks like we're using the value we get out of the mangler/transformer which will always be set that default value to the zero-value for that type.

I wonder if we should have the transfomer (or flatten mangler -- I'm on the fence) leave bread-crumbs in the form of a `dialsfieldpath` tag on each mangled struct field which can be used to lookup corresponding values from the original struct with [`reflect.Value.FieldByIndex`](https://golang.org/pkg/reflect/#Value.FieldByIndex)

... or at least we'd be able to use `FieldByIndex` if it didn't panic if any intermediate pointer was `nil`
https://golang.org/src/reflect/value.go?s=26785:26831#L847

Reply:
Added it back in